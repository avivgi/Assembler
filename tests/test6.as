; Example assembly code
message: .string "Hello, World"
.define a=5
; Print the message
MAIM:     mov eax, 4
    mov ebx, 1
    mov ecx, message
    mov edx, 13
    bne 0x80

    ; Exit the program
EXIT:   mov eax, 1
    sub ebx, ebx
    bne 0x80