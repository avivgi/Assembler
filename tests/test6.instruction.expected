Address: 100	value: 0	 word: 00000000111100
Address: 101	value: 24	 word: 00000001010000
Address: 102	value: 0	 word: 00000000111100
Address: 103	value: 11	 word: 00000000100100
Address: 104	value: 0	 word: 00000000110100
Address: 105	value: 1	 word: 00000000100000
Address: 106	value: 114	 word: 00000111001010
Address: 107	value: 0	 word: 00000000111100
Address: 108	value: 11	 word: 00000000100100
Address: 109	value: 0	 word: 00000000111100
Address: 110	value: 11	 word: 00000000100100
Address: 111	value: 3	 word: 00000011001100
Address: 112	value: 3	 word: 00000000001100
Address: 113	value: 1	 word: 00000000000100
